src/client/pool.rs

```diff
@@ -33,7 +33,7 @@ pub(super) struct Pool<T> {
 // This is a trait to allow the `client::pool::tests` to work for `i32`.
 //
 // See https://github.com/hyperium/hyper/issues/1429
-pub(super) trait Poolable: Unpin + Send + Sized + 'static {
+pub trait Poolable: Unpin + Send + Sized + 'static {
```

Comment:
I'm fine with just making it all public to start with, to unblock people, but I do think the exposed design could be improved, by making the `Pool` basically a `MakeService` over another `MakeService` (a caching layer over the `Connect` stuff).